datasets/noaa-climate-normals/noaa_climate_normals.py

```diff
@@ -0,0 +1,202 @@
+
+import pystac
+import requests
+from stactools.noaa_climate_normals.gridded.constants import (
+    Frequency as GriddedFrequency,
+)
+from stactools.noaa_climate_normals.gridded.constants import Period as GriddedPeriod
+from stactools.noaa_climate_normals.gridded.stac import (
+    create_items as gridded_create_items,
+)
+from stactools.noaa_climate_normals.gridded.utils import nc_href_dict
+from stactools.noaa_climate_normals.netcdf.stac import create_item as netcdf_create_item
+from stactools.noaa_climate_normals.tabular.constants import (
+    Frequency as TabularFrequency,
+)
+from stactools.noaa_climate_normals.tabular.constants import Period as TabularPeriod
+from stactools.noaa_climate_normals.tabular.stac import (
+    create_item as tabular_create_item,
+)
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+from pctasks.core.storage.base import Storage
+
+PARQUET_CONTAINER = "blob://noaanormals/climate-normals-geoparquet"
+COG_CONTAINER = "blob://noaanormals/gridded-normals-cogs"
+NETCDF_STAC_API_URL = "https://planetarycomputer.microsoft.com/api/stac/v1/collections/noaa-climate-normals-netcdf/items/"  # noqa
+
+handler = logging.StreamHandler()
+handler.setFormatter(logging.Formatter("[%(levelname)s]:%(asctime)s: %(message)s"))
+handler.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.INFO)
+
+
+class MissingGeoParquet(Exception):
+    pass
+
+
+class NoaaClimateNormalsTabular(Collection):
+    @classmethod
+    def create_item(
+        cls, asset_uri: str, storage_factory: StorageFactory
+    ) -> Union[List[pystac.Item], WaitTaskResult]:
+
+        uri_parts = asset_uri.split("/")
+        period = TabularPeriod(uri_parts[-3])
+        frequency = TabularFrequency(uri_parts[-4].split("-")[-1])
+
+        csv_storage = storage_factory.get_storage("/".join(uri_parts[:-1]))
+        csv_paths = list(csv_storage.list_files(extensions=[".csv"]))
+
+        parquet_storage = storage_factory.get_storage(PARQUET_CONTAINER)
+        parquet_directory = f"{period.value.replace('-', '_')}-{frequency}.parquet"
+
+        logger.info(
+            f"Processing Climate Normals for Period '{period}', Frequency '{frequency}'"
+        )
+
+        with TemporaryDirectory() as tmp_dir:
+            # download all CSVs, retrying on connection resets and hanging downloads
+            tmp_csv_paths = []
+            failures = True
+            retry_num = 0
+            while failures:
+                num_csvs = len(csv_paths)
+                failed_csv_paths = []
+                for i, csv_path in enumerate(csv_paths, 1):
+                    try:
+                        tmp_csv_path = Path(tmp_dir, csv_path)
+                        auth_url = csv_storage.get_authenticated_url(csv_path)
+                        response = requests.get(auth_url, timeout=10)
```

Comment:
Is this making a request to `http://planetarycomputer.microsoft.com/api/sas/v1`, or do you know if it's using `planetary_computer` internally? That also has retry logic built in (same arguments as the azure libraries).

Reply:
Getting the authenticated url has not surfaced as a problem (yet). It's the file download portion. 

But to your question, I don't _think_ a request is made to `http://planetarycomputer.microsoft.com/api/sas/v1`. I think it is using a storage account key magically derived from the workflow, which can then be used generate a SAS (via the Azure SDK) for use in instantiating a `BlobStorage` class. But I'm out of knowledge zone here.

Reply:
That sounds right to me.